components/base/kinematicbase/differentialDrive.go

```diff
@@ -225,13 +241,31 @@ func (ddk *differentialDriveKinematics) issueCommand(ctx context.Context, curren
 	if err != nil {
 		return false, err
 	}
-	ddk.logger.Debug("distErr: %f\theadingErr %f", distErr, headingErr)
+	ddk.logger.Debugf("distErr: %.2f\theadingErr %.2f", distErr, headingErr)
 	if distErr > ddk.options.GoalRadiusMM && math.Abs(headingErr) > ddk.options.HeadingThresholdDegrees {
 		// base is headed off course; spin to correct
-		return true, ddk.Spin(ctx, math.Min(headingErr, ddk.options.MaxSpinAngleDeg), ddk.options.AngularVelocityDegsPerSec, nil)
+		err := ddk.Spin(ctx, math.Min(headingErr, ddk.options.MaxSpinAngleDeg), ddk.options.AngularVelocityDegsPerSec, nil)
```

Comment:
Same here - should this logic be part of the motion wrapper logic or just moved to explore?

Reply:
Discussing in https://github.com/viamrobotics/rdk/pull/3218#discussion_r1406485369